CONTRIBUTING.md

```diff
@@ -379,12 +379,88 @@ The following limitations are recommended:
 
 ### Observability
 
-Out of the box, your users should be able to observe the state of your component.
-See [observability.md](docs/observability.md) for more details.
+Out of the box, your users should be able to observe the state of your
+component. See [observability.md](docs/observability.md) for more details.
 
 When using the regular helpers, you should have some metrics added around key
-events automatically. For instance, exporters should have `otelcol_exporter_sent_spans`
-tracked without your exporter doing anything.
+events automatically. For instance, exporters should have
+`otelcol_exporter_sent_spans` tracked without your exporter doing anything.
+
+Custom metrics can be defined as part of the `metadata.yaml` for your component.
+The authoritative source of information for this is [the
+schema](https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/mdatagen/metadata-schema.yaml),
```

Comment:
is this consistent with weaver and how telemetry schemas are defined?

Reply:
this is consistent w/ how metrics are configured in other use-cases for mdatagen in the collector contrib for example where scrapers can be configured via metadata.yaml:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/88f8680c4eca59ad844517beed7f60e5d8808444/receiver/redisreceiver/metadata.yaml#L65-L81